From 08bf9ce18f8777de680248ee46ef2e7ce8d84a63 Mon Sep 17 00:00:00 2001
From: Danny Baumann <dannybaumann@web.de>
Date: Thu, 13 Nov 2014 14:31:37 +0100
Subject: [PATCH] Proper supplementary service notification handling (3/5)

Pass parsed call properties around instead bleeding the implementation
detail 'Supplementary service notification' into all layers.

Change-Id: Ib7c3a494024013c261b1d708c47d0e80ae5876dd
---
 src/com/android/server/telecom/ParcelableCallUtils.java | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/src/com/android/server/telecom/ParcelableCallUtils.java b/src/com/android/server/telecom/ParcelableCallUtils.java
index 527e8e6c..21e30737 100644
--- a/src/com/android/server/telecom/ParcelableCallUtils.java
+++ b/src/com/android/server/telecom/ParcelableCallUtils.java
@@ -339,7 +339,13 @@ public class ParcelableCallUtils {
         android.telecom.Call.Details.PROPERTY_SELF_MANAGED,
 
         Connection.PROPERTY_ASSISTED_DIALING_USED,
-        android.telecom.Call.Details.PROPERTY_ASSISTED_DIALING_USED
+        android.telecom.Call.Details.PROPERTY_ASSISTED_DIALING_USED,
+
+        Connection.PROPERTY_WAS_FORWARDED,
+        android.telecom.Call.Details.PROPERTY_WAS_FORWARDED,
+
+        Connection.PROPERTY_REMOTE_INCOMING_CALLS_BARRED,
+        android.telecom.Call.Details.PROPERTY_REMOTE_INCOMING_CALLS_BARRED
     };
 
     private static int convertConnectionToCallProperties(int connectionProperties) {
-- 
2.17.1

